app/models/task.py

```diff
@@ -1,5 +1,50 @@
 from app import db
+from flask import make_response, abort
 
 
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    task_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String)
+    description = db.Column(db.String)
+    completed_at = db.Column(db.DateTime, nullable=True, default=None)
+    goal = db.relationship("Goal", back_populates="tasks")
+    goal_id = db.Column(db.Integer, db.ForeignKey("goal.goal_id"), nullable=True)
+
+    def to_json(self):
+        if self.completed_at:
+            is_complete = True
+        else:
+            is_complete = False
+
+        if self.goal_id:
+            return {
+                "id": self.task_id,
+                "title": self.title,
+                "description": self.description,
+                "is_complete": is_complete,
+                "goal_id": self.goal_id
+            }
+        else:
+            return {
+                "id": self.task_id,
+                "title": self.title,
+                "description": self.description,
+                "is_complete": is_complete
+            }
```

Comment:
This works fine, but you can cut down on a bit of repetition using `bool` and a local variable for your return value:

 ```python
    def to_json(self):
        var json_dict = {
            "id": self.task_id,
            "title": self.title,
            "description": self.description,
            "is_complete": bool(self.completed_at)
        }

        if self.goal_id:
            json_dict["goal_id"] = self.goal_id

        return json_dict
```